From ca7c358d819cc16e4c5ce9a2c2ce187bf32886cd Mon Sep 17 00:00:00 2001
From: Ivan Vecera <ivecera@redhat.com>
Date: Wed, 19 Nov 2014 16:14:38 -0500
Subject: [ethernet] r8169: add disable/enable RTL8411B pll function

Message-id: <1416413687-12219-8-git-send-email-ivecera@redhat.com>
Patchwork-id: 100359
O-Subject: [RHEL7.1 PATCH 07/16] r8169:add disable/enable RTL8411B pll function
Bugzilla: 1165764
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

BZ: #1165764

Upstream commit(s):
commit 42fde7371035144037844f41bd16950de9912bdb
Author: Chun-Hao Lin <hau@realtek.com>
Date:   Wed Oct 1 23:17:14 2014 +0800

    r8169:add disable/enable RTL8411B pll function

    RTL8411B can support disable/enable pll function.

    Signed-off-by: Chun-Hao Lin <hau@realtek.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Ivan Vecera <ivecera@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/realtek/r8169.c b/drivers/net/ethernet/realtek/r8169.c
index 5e84af9..49967c6 100644
--- a/drivers/net/ethernet/realtek/r8169.c
+++ b/drivers/net/ethernet/realtek/r8169.c
@@ -4367,6 +4367,7 @@ static void r8168_pll_power_down(struct rtl8169_private *tp)
  case RTL_GIGA_MAC_VER_31:
  case RTL_GIGA_MAC_VER_32:
  case RTL_GIGA_MAC_VER_33:
+ case RTL_GIGA_MAC_VER_44:
  case RTL_GIGA_MAC_VER_45:
  case RTL_GIGA_MAC_VER_46:
   RTL_W8(PMCH, RTL_R8(PMCH) & ~0x80);
@@ -4394,6 +4395,7 @@ static void r8168_pll_power_up(struct rtl8169_private *tp)
  case RTL_GIGA_MAC_VER_33:
   RTL_W8(PMCH, RTL_R8(PMCH) | 0x80);
   break;
+ case RTL_GIGA_MAC_VER_44:
  case RTL_GIGA_MAC_VER_45:
  case RTL_GIGA_MAC_VER_46:
   RTL_W8(PMCH, RTL_R8(PMCH) | 0xc0);
-- 
1.7.1